### PR TITLE
Remove altair and use pyplot by default

### PR DESCRIPTION
**Pull request by TheCleric**

### CODE REVIEW

**Comment by schorrm** [comment]
Holding back on this one for a little bit to reach agreement / consensus on how to deal with the dependency issues

**Comment by schorrm** [comment]
Leaning towards deprecating Altair

File: pybaseball/plotting.py [hunk 1]
```diff
@@ -1,34 +1,58 @@
...
 
+# Use altair if available for backwards compatibility, else use pyplot
+try:
+    import altair as alt
+    import warnings
+
+    _ALTAIR_ENABLED = True
+    _ALTAIR_WARNING = """
+        Altair functionality is deprecated and will be removed in a future version.
+        Use pyplot functions instead or remove altair the altair library to prevent autodetection.
```
Comment (schorrm) [inline pybaseball/plotting.py:25]: IDK about the end here...

File: pybaseball/plotting.py [hunk 2]
```diff
@@ -1,34 +1,58 @@
...
+    """
+except ImportError:
+    _ALTAIR_ENABLED = False
+
+    # Needed so we can still use altair datatypes for now in our func definitions
+    class FakeAltair():  # pylint: disable=too-few-public-methods
+        Chart = Any
+        LayerChart = Any
+
+    alt = FakeAltair()
```
Comment (schorrm) [inline pybaseball/plotting.py:35]: Do we _need_ this?

File: pybaseball/plotting.py (hunk 1 above)
Comment (TheCleric) [reply pybaseball/plotting.py:25]: I accidentally a word I guess. 🤣

File: pybaseball/plotting.py (hunk 2 above)
Comment (TheCleric) [reply pybaseball/plotting.py:35]: I've cleaned this up so it's less ugly/hacky feeling.

**Comment by schorrm** [comment]
@TheCleric -- pull the CI changes back in and strip the Altair as discussed, and we'll merge?

**Comment by TheCleric** [comment]
@schorrm done.

File: pybaseball/plotting.py [hunk 3]
```diff
@@ -87,40 +102,43 @@ def spraychart(data, team_stadium, title='', tooltips=[], size=100,
...
+    return base
 
-def plot_bb_profile(df: pd.DataFrame, parameter: Optional[str] = "launch_angle"):
+
+def plot_bb_profile(df: pd.DataFrame, parameter: Optional[str] = "launch_angle") -> None:
     """Plots a given StatCast parameter split by bb_type
 
     Args:
-        df (pd.DataFrame): StatCast pd.DataFrame (retrieved through statcast, statcast_batter, etc)
+        df (pd.DataFrame):                   StatCast pd.DataFrame (retrieved through statcast, statcast_batter, etc)
```
Comment (schorrm) [inline pybaseball/plotting.py:134]: nitpick: wacky indent

Comment (TheCleric) [reply pybaseball/plotting.py]: I tried making all the docstrings in this file like the google style + types that @jldbc and I had talked about in #125 . Let me know what you think.

**Comment by schorrm** [comment]
LGTM

**Pull request merged**